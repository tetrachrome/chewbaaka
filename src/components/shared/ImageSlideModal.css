/**
 * ImageSlideModal.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 05, 2020
 * Updated  : Jul 12, 2020
 */

div.ImageSliderModalCoverTriggerOuterContainer {
  position: relative;
  float: left;
  background-color: gray;
  background-size: cover;
  opacity: 0.6;
  border-radius: 6px;
  border: 1px solid gray;
}

div.ImageSliderModalCoverTriggerImageIconInnerContainer {
  position:absolute;
  bottom:0;
  right:0;
}

div.ImageSliderModalCoverTriggerOuterContainer:hover {
  filter: opacity(60%)
}

div.ImageSlideModalComponentModalDescriptionPaginationButtonContainer {
  margin-top: 20px;
}

p.ImageSlideModalComponentModalDescriptionPaginationLabel {
  margin-top: 20px;
  color: rgba(100,100,100,1.0);
  font-style: italic;
  font-weight: 700;
}

div.ImageSlideModalComponentModalContentOuterContainer {
  margin: 0 auto;
}

div.ImageSlideModalComponentModalContentOuterContainer_DebugColor {
  background-color: palegoldenrod;
}

img.ui.huge.image.ImageSlideModalComponentModalContentImage {
  display: block;
  margin: 0 auto;
}

div.ImageSlideModalComponentModalContentImageContainer {
  margin: 0 auto;
}

div.ImageSlideModalComponentModalContentImageContainer_DebugColor {
  background-color: orangered;
}

div.ImageSlideModalComponentModalDescriptionContainer {
  display: block-inline;
  margin: 0 auto;
}

h3.ui.header.ImageSlideModalComponentModalDescriptionHeader {
  margin-top: 20px;
}
