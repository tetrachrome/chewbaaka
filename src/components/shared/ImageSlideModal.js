/**
 * ImageSlideModal.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 05, 2020
 * Updated  : Jul 05, 2020
 */

import React from 'react';

import "semantic-ui-css/semantic.min.css";

import { Button, Header, Image, Icon, Modal } from 'semantic-ui-react'

import "./ImageSlideModal.css"

class ImageSlideModal extends React.Component {

  // TODO: Style this component.

  constructor(props) {
    super(props);
    this.state = {
      isModalOpen: false,
      activeIndex : 0
    }
    this.handleOpen = this.handleOpen.bind(this);
    this.handleClose = this.handleClose.bind(this);
    this.handleOnPrevClick = this.handleOnPrevClick.bind(this);
    this.handleOnNextClick = this.handleOnNextClick.bind(this);
  }

  handleOpen() {
    this.setState((prevState) => ({
      isModalOpen: true,
      activeIndex : prevState.activeIndex
    }));
  }

  handleClose() {
    this.setState((prevState) => ({
      isModalOpen: false,
      activeIndex : 0 // reset to 0
    }));
  }

  handleOnPrevClick() {
    this.setState((prevState) => ({
      isModalOpen: prevState.isModalOpen,
      activeIndex : Math.max(prevState.activeIndex-1, 0)
    }));
  }

  handleOnNextClick() {
    this.setState((prevState) => ({
      isModalOpen: prevState.isModalOpen,
      activeIndex : Math.min(prevState.activeIndex+1, this.props.slides.length - 1)
    }));
  }

  render() {
    const coverImage = this.props.slides[0].image;

    const modalCoverImageContainerDivStyle = {
      width: 480,
      height: 320,
      backgroundImage: `url(${coverImage})`,
    };

    const leftPaginationButtonDisabled = ( this.state.activeIndex === 0 );
    const rightPaginationButtonDisabled = ( this.state.activeIndex ===  this.props.slides.length - 1);

    return (
      <Modal
        trigger={
          <div className="modalCoverImageContainerDiv" style={modalCoverImageContainerDivStyle} onClick={this.handleOpen} data-testid="ImageSlideModalComponentCoverImageContainerDivTestId">
            <div className="modalCoverImageIcon"><Icon name="caret square right outline" color="grey" size="huge" onClick={this.handleOpen} /></div>
          </div>
        }
        open={this.state.isModalOpen}
        onClose={this.handleClose}
        closeIcon
      >
        <Modal.Content image>
          <Image wrapped size='large' src={this.props.slides[this.state.activeIndex].image} />
          <Modal.Description>
            <Header>
              {this.props.slides[this.state.activeIndex].title}
            </Header>
            <p>
              {this.props.slides[this.state.activeIndex].description}
            </p>
            <div className="modalPaginationButtonContainerDiv">
              <Button icon size="small" attached="left" disabled={leftPaginationButtonDisabled} onClick={this.handleOnPrevClick}>
                <Icon name="angle left" size="large" />
              </Button>
              <Button icon size="small" attached="right" disabled={rightPaginationButtonDisabled} onClick={this.handleOnNextClick}>
                <Icon name="angle right" size="large" />
              </Button>
            </div>
            <p className="modalPageLabel">{this.state.activeIndex+1} / {this.props.slides.length}</p>
          </Modal.Description>
        </Modal.Content>
      </Modal>
    )
  }
}

export default ImageSlideModal;
