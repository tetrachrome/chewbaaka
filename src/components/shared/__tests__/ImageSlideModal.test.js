/**
 * ImageSlideModal.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 05, 2020
 * Updated  : Jul 05, 2020
 */

import React from 'react';

import { render, screen } from '@testing-library/react';

import ImageSlideModal from '../ImageSlideModal';

import image01 from '../assets/cheetah-conservation-fund-logo.jpg'
import image02 from '../assets/cheetah-conservation-fund-30-logo.svg'

test('renders ImageSlideModal component', () => {
  render(
    <ImageSlideModal
      slides={[
        {
          image: image01,
          title: "Acinonyx jubatus",
          description: "This is the cheetah."
        },
        {
          image: image02,
          title: "Cheeeeeetah",
          description: "Fastest land animal on earth!"
        }
      ]}
    />
  );

  const componentCoverImageContainerDivPart = screen.getByTestId("ImageSlideModalComponentCoverImageContainerDivTestId");
  expect(componentCoverImageContainerDivPart).toBeInTheDocument();
});
