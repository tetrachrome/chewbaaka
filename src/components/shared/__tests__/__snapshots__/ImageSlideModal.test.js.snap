// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageSlideModal component snapshot 1`] = `
<div
  className="modalCoverImageContainerDiv"
  data-testid="ImageSlideModalComponentCoverImageContainerDivTestId"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(cheetah-conservation-fund-logo.jpg)",
      "height": 320,
      "width": 480,
    }
  }
>
  <div
    className="modalCoverImageIcon"
  >
    <i
      aria-hidden="true"
      className="grey caret square right outline huge icon"
      onClick={[Function]}
    />
  </div>
</div>
`;
